drivers/serial/uart_stm32.c

```diff
@@ -669,10 +671,32 @@ static int uart_stm32_init(struct device *dev)
 	LL_USART_SetTransferDirection(UartInstance,
 				      LL_USART_DIRECTION_TX_RX);
 
-	/* 8 data bit, 1 start bit, 1 stop bit, no parity */
+	/* Determine the parity, if we use other than 'none' we must use 9B data (to get 8 data + 1 parity) */
+	if ( strcmp( config->parity, "even" ) == 0 )
```

Comment:
I'm not quite fond of using string value for this.
I know it is used on other drivers (USB for now), but this isn't really optimized in any way.
On USB driver, it has to be compared to the size of the USB stack, so let's say it ok.
In a serial driver this is more visible.
This being said, it would have to be evaluated to see the real impact

Reply:
I can change this as i clean the code for the checks - would an int 0 (none) 1 (odd) 2(even) be ok?

Or maybe string enum of "N", "E", "O" ? 

I did not clean the commit as i was not sure is it generally even accepted - like i tried to write in the original pull comment.

Reply:
> would an int 0 (none) 1 (odd) 2(even) be ok?

Would be nice to have other opinons, but this is fine for me